internal/locate/region_request.go

```diff
@@ -619,7 +619,9 @@ func (state *accessFollower) isCandidate(idx AccessIndex, replica *replica) bool
 			(!state.option.leaderOnly && (state.tryLeader || idx != state.leaderIdx) && replica.store.IsLabelsMatch(state.option.labels) && (!state.learnerOnly || replica.peer.Role == metapb.PeerRole_Learner)) &&
 				// And If the leader store is abnormal to be accessed under `ReplicaReadPreferLeader` mode, we should choose other valid followers
 				// as candidates to serve the Read request.
-				(!state.option.preferLeader || !replica.store.isSlow()))
+				(!state.option.preferLeader || !replica.store.isSlow())) &&
+		// If the stores are limited, check if the store is in the list.
+		replica.store.IsStoreMatch(state.option.stores)
```

Comment:
The single expression is scarily complicated... I'd prefer split them into multiple variables.